python/ray/ml/train/integrations/sklearn/sklearn_trainer.py

```diff
@@ -112,26 +157,273 @@ def __init__(
-        return super().training_loop()
+        register_ray()
+
+        self.estimator.set_params(**self.params)
+
+        datasets = self._get_datasets()
+        X_train, y_train = datasets.pop(TRAIN_DATASET_KEY)
+        groups = None
+        if "cv_groups" in X_train.columns:
+            groups = X_train["cv_groups"]
+            X_train = X_train.drop("cv_groups", axis=1)
+
+        trial_resources = tune.get_trial_resources().required_resources
+        num_cpus = int(trial_resources.get("CPU", 0))
+        has_gpus = bool(trial_resources.get("GPU", 0))
+
+        # see https://scikit-learn.org/stable/computing/parallelism.html
+        os.environ["OMP_NUM_THREADS"] = str(num_cpus)
+        os.environ["MKL_NUM_THREADS"] = str(num_cpus)
+        os.environ["OPENBLAS_NUM_THREADS"] = str(num_cpus)
+        os.environ["BLIS_NUM_THREADS"] = str(num_cpus)
+
+        parallelize_cv = self._get_cv_parallelism(has_gpus)
+        if self.set_estimator_cpus:
+            num_estimator_cpus = 1 if parallelize_cv else num_cpus
+            set_cpu_params(self.estimator, num_estimator_cpus)
+
+        with parallel_backend("ray", n_jobs=num_cpus):
+            start_time = time()
+            self.estimator.fit(X_train, y_train, **self.fit_params)
+            fit_time = time() - start_time
+
+            with tune.checkpoint_dir(step=1) as checkpoint_dir:
+                with open(os.path.join(checkpoint_dir, MODEL_KEY), "wb") as f:
+                    cpickle.dump(self.estimator, f)
+
+            if self.label_column:
+                validation_set_scores = self._score_on_validation_sets(
+                    self.estimator, datasets
+                )
+                cv_scores = self._score_cv(
+                    self.estimator,
+                    X_train,
+                    y_train,
+                    groups,
+                    # if estimator has parallelism, use that. Otherwise,
+                    # parallelize CV
+                    n_jobs=1 if not parallelize_cv else num_cpus,
+                )
+            else:
+                validation_set_scores = {}
+                cv_scores = {}
+
+        # cv_scores will not override validation_set_scores as we
+        # check for that during initialization
+        results = {
+            **validation_set_scores,
+            **cv_scores,
+            "fit_time": fit_time,
+        }
+        tune.report(**results)
+
+    def as_trainable(self) -> Type[Trainable]:
+        trainable_cls = super().as_trainable()
+        scaling_config = self.scaling_config
+
+        class SklearnTrainable(trainable_cls):
+            @classmethod
+            def default_resource_request(cls, config):
+                updated_scaling_config = config.get("scaling_config", scaling_config)
+                updated_scaling_config = ScalingConfigDataClass(
+                    **updated_scaling_config
+                )
+                ScalingConfigDataClass.validate_config(
```

Comment:
See the comments in the other PR, but I would say let's move this validation to the base Trainer class and have the allowed_keys as class attributes. 

If we do that, and then also remove `_postprocess_checkpoint`, then this subclass should not need to override `as_trainable`. And if it's still necessary, that means we need to rethink our original API.